scripts/Muon/GUI/FrequencyDomainAnalysis/frequency_context.py

```diff
@@ -0,0 +1,102 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import (absolute_import, division, print_function)
+
+import mantid.simpleapi as mantid
+
+import Muon.GUI.Common.ADSHandler.workspace_naming as wsName
+
+class FrequencyContext(object):
+    """
+    A simple class for identifing the current run
+    and it can return the name, run and instrument.
+    The current run is the same as the one in MonAnalysis
+    """
+
+    def __init__(self, context):
+        self.context = context
+        self.options = "None"
+        self.options = [item.replace(" ", "") for item in self.options]
+        self.N_points = 1
+        self.instrument = "None"
+
+        self.runName = "None"
+
+    @property
+    def version(self):
+        return 2
+
+    def setUp(self, tmpWS):
+        # get everything from the ADS
```

Comment:
can't you get most to all of this information from the current contex rather than going via the AWS?